index.js

```diff
@@ -37,41 +37,85 @@ var EchoClient = function(config){
  * @param {object} data.props Other properties associated with the event
  * @callback callback
  */
-EchoClient.prototype.addEvents = function(token, data, callback){
+ EchoClient.prototype.addEvents = function(token, data, callback){
+     if(!token){
+         throw new Error('Missing Persona token');
+     }
+
+     if(!data){
+         throw new Error('Missing data');
+     }
+
+     // multiple events can be written by posting an array
+     // if not an array, check data looks ok
+     if (!(data instanceof Array)) {
+         if(!data.class){
+             throw new Error('Missing field data.class');
+         }
+         if(!data.source){
+             throw new Error('Missing field data.source');
+         }
+     }
+
+     var requestOptions = {
+         url: this.config.echo_endpoint + '/1/events',
+         headers: {
+             'Accept': 'application/json',
+             'Authorization': 'Bearer ' + token
+         },
+         body: data,
+         method: 'POST',
+         json: true
+     };
+
+     request.post(requestOptions, function(err, response, body){
+         if(err){
+             callback(err);
+         } else{
+             callback(null, body);
+         }
+     });
+
+     this.debug(JSON.stringify(requestOptions));
+ };
+
+/**
+ * Query analytics using a passed path
+ * @param  {string}   token         Persona token
+ * @param  {string}   queryPath     URL path to query
+ * @param  {boolean}  useCache      Indicates if cache should be used or not
+ * @callback callback
+ */
+EchoClient.prototype.queryAnalytics = function(token, queryPath, useCache, callback) {
     if(!token){
         throw new Error('Missing Persona token');
     }
-    if(!data){
-        throw new Error('Missing data');
-    }
-    if(!data.class){
-        throw new Error('Missing field data.class');
-    }
-    if(!data.source){
-        throw new Error('Missing field data.source');
+    if(!queryPath){
+        throw new Error('Missing Analytics queryPath');
     }
 
     var requestOptions = {
-        url: this.config.echo_endpoint+'/1/events',
+        url: this.config.echo_endpoint + '/1/analytics/' + queryPath,
+        method: 'GET',
         headers: {
-            'Accept': 'application/json',
-            'Authorization':'Bearer '+token
-        },
-        body:data,
-        method:'POST',
-        json:true
+            'Content-Type': 'application/json',
+            'Authorization': 'Bearer ' + token
+        }
     };
 
-    request.post(requestOptions, function(err, response, body){
-        if(err){
+    if (useCache === false) {
+        requestOptions.headers['cache-control'] = 'none';
+    }
+
+    request.get(requestOptions, function(err, response, body) {
+        if (err) {
             callback(err);
-        } else{
+        } else {
             callback(null, body);
         }
     });
 
     this.debug(JSON.stringify(requestOptions));
```

Comment:
Again, move this before the callback is called, it will make more sense looking in the log file intermingled with other log lines.